lib/pingurl.js

```diff
@@ -24,12 +24,20 @@ exports.ping = (pingUrl, inputData, timeout, timeoutWait) => {
     let pinger = new HttpPing(timeout, timeoutWait);
     let opts = {
       url: pingUrl,
-      host: parsed.host,
+      host: parsed.hostname,
       path: parsed.path,
       port: parsed.port,
       headers: exports.parseHeaders(inputData)
     };
-    pinger.get(opts, resolve, reject);
+
+    let failsafe = setTimeout(function() {
+      reject(new Error(`HTTP timeout failsafe ${pingUrl}`));
+    }, pinger.timeout + 1000);
```

Comment:
so this times out 1 sec after the pinger should time out? i.e. in case it doesn't?